src/query/executor/transform/lazy.go

```diff
@@ -86,25 +88,42 @@ func (s *stepIter) StepCount() int {
 }
 
 func (s *stepIter) Next() bool {
-	return s.iter.Next()
+	if s.err != nil {
+		return false
+	}
+
+	next := s.iter.Next()
+	if !next {
+		s.err = s.iter.Err()
```

Comment:
Hm, instead of adding this here, perhaps in `Err()` we can instead just return it as well when it's asked for?  Just otherwise this logic is just a little bit more difficult to follow (also if the inner iterator gets an error for some other reason later, it will properly propagate, unlike here where it can only get set during a call to next?):

```go
func (s *stepIter) Err() error {
	if s.err != nil {
		return s.err
	}
	return s.iter.Err()
}
```

Reply:
Nice one, will do 👍